### PR TITLE
Fix llama and gemma modeling patching for openvino export

### PR DESCRIPTION
**Pull request by echarlaix**
Fix compatibility for transformers v4.41.0 

cc @helena-intel @eaidova

### CODE REVIEW

File: optimum/exporters/openvino/model_configs.py [hunk 1]
```diff
@@ -90,6 +91,21 @@ def init_model_configs():
 register_in_tasks_manager = TasksManager.create_register("openvino", overwrite_existing=True)
 
 
+@register_in_tasks_manager(
```
Comment (eaidova) [inline optimum/exporters/openvino/model_configs.py:94]: I do not see any difference with onnx version config. Why do you need to have it explicitly?

Comment (echarlaix) [reply optimum/exporters/openvino/model_configs.py:94]: Added as `DEFAULT_ONNX_OPSET` now needs to be set to 14 (now that F.scaled_dot_product_attention is used by default for this model), will be removed before merging the PR as it'll be added in optimum directly

File: optimum/intel/openvino/trainer.py [hunk 2]
```diff
@@ -153,8 +153,6 @@
         "{re}.*Embedding.*",
         "{re}.*add___.*",
         "{re}.*layer_norm_.*",
-        "{re}.*matmul_1",
-        "{re}.*__truediv__.*",
     ],
```
Comment (echarlaix) [inline optimum/intel/openvino/trainer.py:156]: Removed to fix 
```
nncf.errors.ValidationError: No match has been found among the model operations for the following ignored/target scope definitions: - ignored_scope: ['{re}.*__truediv__.*', '{re}.*matmul_1']
```

Comment (echarlaix) [reply optimum/intel/openvino/trainer.py:156]: not related to llama and gemma modeling patching so moving it to https://github.com/huggingface/optimum-intel/pull/724

**Pull request merged**